src/Fantomas/CodePrinter.fs

```diff
@@ -4373,12 +4378,32 @@ and genClause astContext hasBar (Clause (p, e, eo) as ce) =
         +> fun ctx -> leaveNodeTokenByName (arrowRange ctx) RARROW ctx
 
     genTriviaBeforeClausePipe p.Range
-    +> ifElse
-        hasBar
-        (sepBar +> atCurrentColumnWithPrepend pat body
```

Comment:
I remove `atCurrentColumnWithPrepend` to solve https://github.com/fsprojects/fantomas/issues/1501 and because it simplifies the flow of the code. 
The only reason we had this was because one nitpick of a user in a time where style guides weren't a thing yet.
When I looked it up in the style I noticed our current behavior was inconsistent.

`AtCurrentColumn` will lock the column indent and override the normal flow of indentation, it is a last resort and that is the main reason I wanted to get rid of it.